lib/lookup/src/lookup_v2/owned.rs

```diff
@@ -262,6 +262,17 @@ impl From<OwnedValuePath> for String {
     }
 }
 
+impl Display for OwnedSegment {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        // This isn't the most efficient method, but it re-uses existing code
+        // Displaying is likely not in a hot path
```

Comment:
I don't love that its "likely", but I'm also failing to think of where it may be used that performance sensitive